Documentation

SoorajModi opened the pull request:
**Describe the pull request:**
<!-- Include a description of the bug/feature and how you solved it -->

- Added a license
- Added documentation for the app including how to run, how to update it, contributing, and accessibility features
- Updated the README

Let me know if there is anything else I should include in the docs.

---

<!-- Tags (add as many as applicable) -->

Closes: #10 <!-- number of issue or pull request -->\
Fixes: # <!-- number of issue (implies Closes tag) or commit SHA -->\
Related: #42 <!-- number of issue/pull request, or link to external discussion -->

---

**Checklist:**

<!-- To check an item, fill the brackets with the letter `x`; the result should look like `[x]`.-->

- [x] Verify the changes work as expected on at least two browsers
- [x] Verify the changes work as expected on mobile view
- [x] Update documentation / not applicable
- [x] Set yourself as the `Assignee`
- [ ] Set `SoorajModi` as the reviewer

MarshallAsch commented on docs/Contributing.md:
```diff
@@ -0,0 +1,63 @@
+# Contributing
```
rename to `CONTRIBUTING.md` (caps) and link to this file from the README

SoorajModi replied:
I wanted to keep all the docs in the same format with camel case titles, for consistency. I know its not how its typically done, but I like it a bit better this way.

Added the link

MarshallAsch replied:
I dont think the link is present anymore

MarshallAsch commented on docs/Accessibility.md:
```diff
@@ -0,0 +1,15 @@
+
+## Color
+
+The primary colors used in this website is `#660000`, `#F5F5F5`, `#465052`, `#FFFFFF`, and `#000000`.
```
Thoughts on adding something like this https://github.com/olizilla/swatches to include color swatches rather than just color codes?

SoorajModi replied:
Did not know this was a thing, added now!

MarshallAsch commented on docs/Contributing.md:
```diff
@@ -0,0 +1,63 @@
+## Code of Conduct
+
+We take our open source community seriously and hold ourselves and other contributors to high standards of communication. 
+By participating and contributing to this project, you agree to uphold our [Code of Conduct](https://github.com/Botomir/Botomir/blob/main/CODE_OF_CONDUCT.md).
```
Code of conduct should be in this project.

SoorajModi replied:
Added

MarshallAsch commented on docs/HowToRun.md:
```diff
@@ -0,0 +1,59 @@
+# How to Run
```
Thoughts on having this in the README instead? Personally I generally only want to have to look at the readme to find this information. I don't want to have to go searching through the project for docs like this.

SoorajModi replied:
Normally, I would agree, however I want to approach this from the perspective of a student who has no web development experience. I want to have extensive instructions on how to run the app locally on each OS. My hope is that future students are able to make changes without our help.

For that reason I made it into its own separate file so we can lots of detail later on. Dont wanna include it in the README, because it might look ugly lol.

MarshallAsch commented on docs/HowToUpdate.md:
```diff
@@ -0,0 +1,65 @@
+# How to Update
+
+This document outlines how to update parts of the website. See `Contributing.md` to see how to contribute to the repo. 
```
make `Contributing.md` a link to the file and not a code snippet

SoorajModi replied:
Fixed

MarshallAsch commented on docs/HowToUpdate.md:
```diff
@@ -0,0 +1,65 @@
+Open `execList.js`, find the individual and either comment them out or delete their section. Also delete their image in
+`src/pages/about/assets`.
+
+## Policy documents
```
Where did these policy docs come from? (google drive), we should mention that so the source does not get forgotten about

SoorajModi replied:
Good call

MarshallAsch reviewed (changes requested):
Looks good, a couple comments then its good.

MarshallAsch commented on README.md:
```diff
@@ -1,79 +1,19 @@
-You will also see any lint errors in the console.
-
-`yarn test`
+![CI testing](https://github.com/UoGSOCIS/socis_website/workflows/CI/badge.svg)
```
I still think we should add some links here to the docs stuff. looking at a git repo people will normally just look at the readme and not go searching for things, I have never assumed that the `docs` folder would hold docs for me as a potential contributor, but rather I often assume that they are docs that go along with the project.

SoorajModi replied:
To be fair does anyone read the documentation XD. But as an alternative, what if we put everything in the wiki instead? Might be the option that makes the most sense, and include very basic instructions on how to run in the README?

MarshallAsch replied:
I agree wiki might be good for some of that, as long as there is at least a link to it from the readme, basic instructions in readme + more detail in wiki  / docs works too (that is my preferred method). I really like readme's that include a quickstart then detailed usage is linked elsewhere

SoorajModi commented:
- Removed the docs folder, and created a wiki instead
- Renamed `CodeOfConduct.md` to `CODE_OF_CONDUCT.md` and moved to root level
- Renamed `Contributing.md` to `CONTRIBUTING.md` and moved to root level
- Added quick start instructions to `README`
- Updated links in `CONTRIBUTING`, note: some of them wont work until the branch is merged with master, let me know if I missed one though
